### PR TITLE
Adding scikit-matter

### PR DESCRIPTION
**Pull request by PicoCentauri**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

File: recipes/scikit-matter/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+  noarch: python
+  number: 0
+
+requirements:
+  host:
+    - python >=3.8
+    - pip
+  run:
+    - python >=3.8
+    - scikit-learn >=1.1.0
```
Comment (dopplershift) [inline recipes/scikit-matter/meta.yaml:23]: Upstream also lists numpy (and uses it directly):
```suggestion
    - python >=3.8
    - numpy
    - scikit-learn >=1.1.0
```

**Review by dopplershift — COMMENTED** [review]
Minor improvement suggestion. I also need confirmation from @Luthaf that they are willing to be listed as a maintainer for this recipe.

**Comment by Luthaf** [comment]
>  I also need confirmation from @Luthaf that they are willing to be listed as a maintainer for this recipe.

I confirm I'm willing to help maintain this recipe!

**Comment by PicoCentauri** [comment]
> Minor improvement suggestion.

Sure, we can add Numpy explicitly here. However, it is a core dependency of scikit-learn and for this pure Python package I am not sure if this is really necessary.

**Comment by dopplershift** [comment]
@PicoCentauri In general, it's better to explicitly list the things that are needed rather than rely on them to be pulled in transitively from other dependencies--it eliminates those packages changing and breaking you (admittedly not a huge risk with numpy and scikit-learn).

**Review by dopplershift — APPROVED** [review]

**Pull request merged**